NuclearData/mgNeutronData/baseMgNeutron/baseMgNeutronDatabase_class.f90

```diff
@@ -124,13 +127,16 @@ function getMajorantXS(self, p) result(xs)
     class(baseMgNeutronDatabase), intent(inout) :: self
     class(particle), intent(in)                 :: p
     real(defReal)                               :: xs
-    integer(shortInt)                           :: i, idx
+    character(100), parameter :: Here = ' getMajorantXS (baseMgNeutronDatabase_class.f90)'
 
-    xs = ZERO
-    do i=1,size(self % activeMats)
-      idx = self % activeMats(i)
-      xs = max(xs, self % getTotalMatXS(p, idx))
-    end do
```

Comment:
I think it may be a considerable problem that in the initialisation active materials are not considered. 
Unless I am missing something (quite possible -> Friday evening ;-) ) the pre-calculated majorant is based on all defined materials, not only the materials which are actually included in the geometry.    

I would suggest moving the majorant calculation to the `activate` subroutine.

Reply:
Very good point.  I will do that!